### ISSUE TITLE
Add new external stream type to write to clickhouse

### ISSUE THREAD

**Issue by jovezhong** [timeplus-io/proton#439]
**Use case**

Proton is a unified data platform for streaming and historical data, powered by ClickHouse. It will be even better if Proton users can send the streaming SQL results to a local or remote ClickHouse (including ClickHouse Cloud), so that the user can choose to run OLAP workload on ClickHouse.

**Describe the solution you'd like**

```sql
CREATE EXTERNAL STREAM write_to_ch (uid int, message string ..)
SETTINGS type='clickhouse', url='http://localhost:8123', 
                   user='clickhouse', password='123', 
                   database='default', table='tab1'
```

We need to provide&document a mapping for Proton data type and ClickHouse data type.

**Describe alternatives you've considered**

Proton can also write data to Kafka topics then use KafkaEngine in ClickHouse or ClickPipe to read them in ClickHouse.
But this will add more components in the pipeline. 
We prefer having a out-of-box integration for Proton and ClickHouse.

**Additional context**

**Comment by jovezhong** [comment]
(Jove Github Bot) assuming it is not done, deferred this ticket to the next sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) assuming it is not done, deferred this ticket to the next sprint.

**Comment by jovezhong** [comment]
Released in 1.4.2 #546
